tests/golos.publication_tests.cpp

```diff
@@ -0,0 +1,439 @@
+
+BOOST_FIXTURE_TEST_CASE(test_create_post_and_battery_limit_post, golos_publication_tester) try {
+    BOOST_REQUIRE_EQUAL(success(), create_battery(N(golos.pub)));
+    produce_blocks(1);
+
+    auto account_battery = get_battery(N(golos.pub));
+    require_equal_battery(account_battery, mvo()
+                          ("posting_battery", mvo()("charge", 0))
+                          ("battery_of_votes", mvo()("charge", 0))
+                          ("limit_battery_posting", mvo()("charge", 0))
+                          ("limit_battery_comment", mvo()("charge", 0))
+                          ("limit_battery_of_votes", mvo()("charge", 0))
+                          );
+
+    BOOST_REQUIRE_EQUAL(success(), golos_publication_tester::create_post(
+                            N(golos.pub),
+                            "permlink0",
+                            N(),
+                            "",
+                            333,
+                            "payouttype",
+                            {{N(beneficiary), 777}},
+                            "paytype",
+                            "headerpost",
+                            "bodypost",
+                            "languagepost",
+                            {{"tag"}},
+                            "jsonmetadata"));
+
+    account_battery = get_battery(N(golos.pub));
+    require_equal_battery(account_battery, mvo()
+                          ("posting_battery", mvo()("charge", 2500))
+                          ("battery_of_votes", mvo()("charge", 0))
+                          ("limit_battery_posting", mvo()("charge", 10000))
+                          ("limit_battery_comment", mvo()("charge", 0))
+                          ("limit_battery_of_votes", mvo()("charge", 0))
+                          );
+    produce_blocks(10);
+
+    BOOST_REQUIRE_EQUAL(error("assertion failure with message: Battery overrun"), golos_publication_tester::create_post(
+                            N(golos.pub),
+                            "permlink1",
+                            N(),
+                            "",
+                            333,
+                            "payouttype",
+                            {{N(beneficiary), 777}},
+                            "paytype",
+                            "headerpost",
+                            "bodypost",
+                            "languagepost",
+                            {{"tag"}},
+                            "jsonmetadata"));
+
+    account_battery = get_battery(N(golos.pub));
+    require_equal_battery(account_battery, mvo()
+                          ("posting_battery", mvo()("charge", 2500))
+                          ("battery_of_votes", mvo()("charge", 0))
+                          ("limit_battery_posting", mvo()("charge", 10000))
+                          ("limit_battery_comment", mvo()("charge", 0))
+                          ("limit_battery_of_votes", mvo()("charge", 0))
+                          );
+    produce_blocks(21);
```

Comment:
21 is magic number, change it to some constant/define, so it easy to understand, what is it